src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.managed.cs

```diff
@@ -830,11 +835,24 @@ private void ClearLines(bool clearContainer)
 				this.Log().LogDebug($"{GetMethodTag()}");
 			}
 
-			foreach (var line in _materializedLines)
+			_clearingLines = true;
+			try
 			{
-				RecycleLine(line, clearContainer);
+				// Recycling a line can lead to removing elements from the visual tree.
+				// If a user forces a layout during this (e.g. in Unloaded), we will
+				// update the _materializedLines in MeasureOverride. So, to prevent
+				// modification while looping, we take a copy and clear before looping.
+				var lines = _materializedLines.ToList();
+				_materializedLines.Clear();
+				foreach (var line in lines)
+				{
+					RecycleLine(line, clearContainer);
+				}
```

Comment:
```suggestion
				// Recycling a line can lead to elements removal from the visual tree.
				// To prevent modification while looping, we take a copy of _materializedLines
				// and clear the original before starting the loop. This ensures that if a user forces 
				// a layout during this process (e.g., in Unloaded), we can update _materializedLines 
				// in MeasureOverride without issues.
				var lines = _materializedLines.ToArray();
				_materializedLines.Clear();
				foreach (var line in lines)
				{
					RecycleLine(line, clearContainer);
				}
```